sugar.go

```diff
@@ -61,6 +61,16 @@ func (s *SugaredLogger) Named(name string) *SugaredLogger {
 	return &SugaredLogger{base: s.base.Named(name)}
 }
 
+// WithOptions clones the base Logger, applies the supplied Options, wraps
+// the resulting Logger, and returns it. It's safe to use concurrently.
```

Comment:
I think this is reporting too much of the implementation.
We can simplify this:

```suggestion
// WithOptions clones the current SugaredLogger, applies the supplied Options,
// and returns the result. It's safe to use concurrently.
```